From aaea3035d8f9af2a73724f871d303b52a8ef8b6f Mon Sep 17 00:00:00 2001
From: Nectarine <nectarine@localhost>
Date: Sat, 24 Oct 2026 14:47:46 -0400
Subject: [PATCH 16/29] win32u: Enable host Vulkan portability enumeration.

From Gcenx macports-wine (0001-win32u-Enable-host-Vulkan-portability-enumeration.diff):
lets the Vulkan loader expose MoltenVK, a portability driver, to Wine.
---
 dlls/win32u/vulkan.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/win32u/vulkan.c b/dlls/win32u/vulkan.c
index f5bc8f93d89..1c6c30d0e60 100644
--- a/dlls/win32u/vulkan.c
+++ b/dlls/win32u/vulkan.c
@@ -1002,6 +1002,11 @@ static VkResult convert_instance_create_info( struct mempool *pool, VkInstanceCr
         instance->obj.extensions.has_VK_EXT_surface_maintenance1 = 1;
     if (vulkan_funcs.host_extensions.has_VK_KHR_get_physical_device_properties2)
         instance->obj.extensions.has_VK_KHR_get_physical_device_properties2 = 1;
+    if (vulkan_funcs.host_extensions.has_VK_KHR_portability_enumeration)
+    {
+        instance->obj.extensions.has_VK_KHR_portability_enumeration = 1;
+        info->flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
+    }
     if (use_external_memory())
         instance->obj.extensions.has_VK_KHR_external_memory_capabilities = 1;
 
-- 
2.43.0

